File: haskell/armstrong-numbers/src/ArmstrongNumbers.hs
module ArmstrongNumbers (armstrong) where

armstrong :: Integral a => a -> Bool
armstrong number = number == s
  where
    d = digits number
    l = length d
    s = sum . map (^l) $ d

digits :: Integral a => a -> [a]
digits = go []
  where
    go acc n
      | n > 0     = go (n `mod` 10:acc) (n `div` 10)
      | otherwise = acc

File: typescript/armstrong-numbers/armstrong-numbers.ts
class ArmstrongNumbers {
    static isArmstrongNumber(num: number): boolean {
        const digits = num.toString().split('')
        const len = digits.length
        const armstrong = digits.reduce((acc, digit) => acc + parseInt(digit) ** len, 0)

        return armstrong == num
    }
}

export default ArmstrongNumbers
